R/sem.R

```diff
@@ -1325,6 +1652,335 @@ checkLavaanModel <- function(model, availableVars) {
+          for (row in row_idx_alpha)
+            alphadf[row, mod_idx + 2] <- c(alphalist[[mod_idx]][grp, which(names(alphalist[[mod_idx]]) == alphadf[row, "factor"])])
+          row_idx_omega <- grp_idx_omega[match(names(omegalist[[mod_idx]]), omegadf[grp_idx_omega, "factor"])]
+          for (row in row_idx_omega)
+            omegadf[row, mod_idx + 2] <- c(omegalist[[mod_idx]][grp, which(names(omegalist[[mod_idx]]) == omegadf[row, "factor"])])
+        }
+      }
+
+      # fill jasp table with data
+      reliabilitytab[["group"]] <- alphadf[["group"]]
+      reliabilitytab[["factor"]] <- alphadf[["factor"]]
+      for (i in seq_along(alphalist)) reliabilitytab[[paste0("reliabilityAlpha_", i)]] <- alphadf[[i + 2]]
+      for (i in seq_along(omegalist)) reliabilitytab[[paste0("reliabilityOmega_", i)]] <- omegadf[[i + 2]]
+    }
+  }
+}
+
+.semHtmt <- function(modelContainer, dataset, options, ready) {
+  if (!options[["htmt"]] || !is.null(modelContainer[["htmt"]])) return()
+
+
+  htmt <- createJaspContainer()
+  htmt$position <- 0.95
+  htmt$dependOn(c("htmt", "naAction", "models"))
+
+  modelContainer[["htmt"]] <- htmt
+
+  if (length(options[["models"]]) < 2) {
+    .semHtmtTables(modelContainer[["results"]][["object"]][[1]], NULL, htmt, options, ready, dataset)
+  } else {
+    htmt$title <- gettext("Heterotrait-monotrait ratio")
+
+    for (i in seq_along(options[["models"]])) {
+      fit <- modelContainer[["results"]][["object"]][[i]]
+      model <- options[["models"]][[i]]
+      .semHtmtTables(fit, model, htmt, options, ready, dataset)
+    }
+  }
+}
+
+.semHtmtTables <- function(fit, model, parentContainer, options, ready, dataset) {
+  if (is.null(model)) {
+    htmtcont <- parentContainer
+    title <- gettext("Heterotrait-monotrait ratio")
+  } else {
+    htmtcont <- createJaspContainer(model[["name"]], initCollapsed = TRUE)
+    title <- ""
+  }
+
+  htmttab <- createJaspTable(title = title)
+  htmtcont[["htmttab"]] <- htmttab
+
+  if (options[["group"]] == "") {
+    lavopts <- .semOptionsToLavOptions(options, dataset)
+    lavmodel <- ifelse(is.null(model), .semTranslateModel(options[["models"]][[1]][["syntax"]], dataset), .semTranslateModel(model[["syntax"]], dataset))
+
+    parTable <- lavaan::lavaanify(lavmodel)
+    latents  <- parTable[parTable$op == "=~",]
+    higherOrder <- unique(latents[!latents$rhs %in% names(dataset),]$lhs)
+    lavmodel <- parTable[!parTable$lhs %in% higherOrder, ]
+
+    if (options[["dataType"]] == "raw") {
+      htmt_result <- semTools::htmt(model = lavmodel, data = dataset, missing = lavopts[["missing"]])
+    } else {
+      htmt_result <- semTools::htmt(model = lavmodel, sample.cov = .semDataCovariance(dataset, model), missing = lavopts[["missing"]])
+    }
+    htmt_result[upper.tri(htmt_result)] <- NA
+
+    for (i in 1:ncol(htmt_result)) {
+      name <- colnames(htmt_result)[i]
+      htmttab$addColumnInfo(name, title = gettext(name), type ="pvalue")
```

Comment:
```suggestion
      htmttab$addColumnInfo(name, title = gettext("name"), type ="pvalue")
```

A typo? yaybe it caused the translation check to fail